src/transaction/transaction_master_group_complete_manager.cpp

```diff
@@ -125,22 +142,27 @@ namespace cubtx
   //
   // prepare_complete prepares group complete. Always should be called before do_complete.
   //
-  void master_group_complete_manager::prepare_complete (THREAD_ENTRY *thread_p)
+  void master_group_complete_manager::do_prepare_complete (THREAD_ENTRY *thread_p)
   {
     if (close_current_group ())
       {
-	cubstream::stream_position closed_group_stream_start_position, closed_group_stream_end_position;
+	cubstream::stream_position closed_group_stream_start_position = 0ULL, closed_group_stream_end_position = 0ULL;
 	const tx_group &closed_group = get_latest_closed_group ();
 
 	/* TODO - Introduce parameter. For now complete group MVCC only here. Notify MVCC complete. */
-	log_Gl.mvcc_table.complete_group_mvcc (closed_group);
+	log_Gl.mvcc_table.complete_group_mvcc (thread_p, closed_group);
 	notify_group_mvcc_complete (closed_group);
 
 	/* Pack group commit that internally wakeups senders. Get stream position of group complete. */
-	logtb_get_tdes (thread_p)->get_replication_generator ().pack_group_commit_entry (closed_group_stream_start_position,
-	    closed_group_stream_end_position);
+	/* TODO - fix pack_group_commit_entry */
+	logtb_get_tdes (thread_p)->get_replication_generator ().pack_group_commit_entry (
+		closed_group_stream_start_position, closed_group_stream_end_position);
 	m_latest_closed_group_start_stream_position = closed_group_stream_start_position;
 	m_latest_closed_group_end_stream_position = closed_group_stream_end_position;
+	mark_latest_closed_group_prepared_for_complete ();
+
+	/* TODO Wakeup senders, just to be sure. */
+	//cubreplication::replication_node_manager::get_master_node()->wakeup_transfer_senders (closed_group_stream_end_position);
```

Comment:
Remove comment?

Reply:
No, it should be activated. I will remove it for now to be clear for reviewers and hope to not forget at future merge.